packages/nimble-components/src/rich-text-viewer/index.ts

```diff
@@ -11,7 +18,100 @@ declare global {
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: fit-to-content
+     */
+    @attr({ attribute: 'fit-to-content', mode: 'boolean' })
+    public fitToContent = false;
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: uses the parsing logic to convert the input markdown string to render in the component.
+     */
+    public set markdownValue(value: string) {
+        this._markdownValue = value;
+        this.serializedContent = this.parseMarkdownToDOM(value);
+        this.appendSerializedContentToViewer();
+    }
+
+    /**
+     * @public
+     * @remarks
+     * Accessor: gets the raw markdown string.
+     */
+    public get markdownValue(): string {
+        return this._markdownValue;
+    }
+
+    private _markdownValue = '';
+    private serializedContent?: HTMLElement | DocumentFragment;
+    private readonly markdownParser: MarkdownParser;
+    private readonly domSerializer: DOMSerializer;
+
+    public constructor() {
+        super();
+        this.domSerializer = DOMSerializer.fromSchema(schema);
+        this.markdownParser = this.initializeMarkdownParser();
+    }
+
+    /**
+     * @internal
+     */
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.appendSerializedContentToViewer();
+    }
+
+    private initializeMarkdownParser(): MarkdownParser {
+        /**
+         * It configures the tokenizer of the default Markdown parser with the 'zero' preset.
+         * The 'zero' preset is a configuration with no rules enabled by default to selectively enable specific rules.
+         * https://github.com/markdown-it/markdown-it/blob/master/lib/presets/zero.js#L1
+         *
+         */
+        const zeroTokenizerConfiguration = defaultMarkdownParser.tokenizer.configure('zero');
+
+        // The detailed information of the supported rules were provided in the below CommonMark spec document.
+        // https://spec.commonmark.org/0.30/
+        const supportedTokenizerRules = zeroTokenizerConfiguration.enable([
+            'emphasis',
+            'list',
+            'autolink'
+        ]);
+
+        return new MarkdownParser(
+            schema,
+            supportedTokenizerRules,
+            defaultMarkdownParser.tokens
+        );
+    }
+
+    /**
+     *
+     * This function takes a markdown string, parses it using the ProseMirror MarkdownParser, serializes the parsed content into a
+     * DOM structure using a DOMSerializer, and returns the serialized result.
+     *
+     */
+    private parseMarkdownToDOM(value: string): HTMLElement | DocumentFragment {
+        return this.domSerializer.serializeFragment(
+            this.markdownParser.parse(value)!.content
+        );
+    }
+
+    private appendSerializedContentToViewer(): void {
+        const viewer = this.shadowRoot?.querySelector('#viewer');
+        if (viewer && this.serializedContent) {
+            viewer.innerHTML = '';
```

Comment:
Is [removeChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild) or [replaceChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild) a better alternative here?

Just to ensure any associated event listeners of the existing elements are cleared reliably.

Reply:
Yes, I have used [replaceChildren](https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren) to replace all the child elements with the parsed node.